src/app/group/[groupId]/join/page.tsx

```diff
@@ -0,0 +1,100 @@
+'use client';
+
+import { notFound, useRouter } from 'next/navigation';
+import { SubmitHandler, useForm } from 'react-hook-form';
+
+import useJoinBookGroup from '@/hooks/group/useJoinBookGroup';
+
+import SSRSafeSuspense from '@/components/SSRSafeSuspense';
+import Loading from '@/v1/base/Loading';
+import Input from '@/v1/base/Input';
+import InputLength from '@/v1/base/InputLength';
+import ErrorMessage from '@/v1/base/ErrorMessage';
+import BottomActionButton from '@/v1/base/BottomActionButton';
+import BookGroupNavigation from '@/v1/bookGroup/BookGroupNavigation';
+
+type JoinFormValues = {
+  answer: string;
+};
+
+const JoinBookGroupPage = ({
+  params: { groupId },
+}: {
+  params: { groupId: number };
+}) => {
+  return (
+    <SSRSafeSuspense fallback={<Loading fullpage />}>
+      <BookGroupNavigation groupId={groupId}>
+        <BookGroupNavigation.BackButton
+          href={`/group/${groupId}`}
+          routeOption="replace"
+        />
+        <BookGroupNavigation.Title />
+      </BookGroupNavigation>
+      <BookGroupJoinForm groupId={groupId} />
+    </SSRSafeSuspense>
+  );
+};
+
+const BookGroupJoinForm = ({ groupId }: { groupId: number }) => {
+  const router = useRouter();
+  const { isMember, hasPassword, question, joinBookGroup } =
+    useJoinBookGroup(groupId);
+
+  if (isMember || !hasPassword) {
+    notFound();
+  }
+
+  const {
+    register,
+    watch,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<JoinFormValues>({ mode: 'all' });
+
+  const submitJoinForm: SubmitHandler<JoinFormValues> = ({ answer }) => {
+    joinBookGroup({
+      answer,
+      onSuccess: () => router.replace(`/group/${groupId}`),
+    });
+  };
+
+  return (
+    <form
+      className="mt-[2.5rem] flex flex-col gap-[2.5rem]"
+      onSubmit={handleSubmit(submitJoinForm)}
+    >
+      <p className="whitespace-pre-line text-2xl font-bold leading-snug">
+        {`문제를 맞추면
+          모임에 가입할 수 있어요`}
+      </p>
+      <div className="flex flex-col gap-[1.5rem]">
+        <p className="text-sm">{question}</p>
+        <div className="flex flex-col gap-[0.5rem]">
+          <Input
+            {...register('answer', {
+              required: '정답을 입력해주세요',
+              minLength: { value: 1, message: '2자 이상 입력해주세요.' },
+              maxLength: { value: 10, message: '10자 이하 입력해주세요.' },
+            })}
+            placeholder="띄어쓰기 없이 정답을 입력해주세요"
+            error={!!errors.answer}
+          />
```

Comment:
p3;

```suggestion
          <Input
            {...register('answer', {
              required: '정답을 입력해주세요',
              pattern: {
                value: /^\S*$/g,
                message: '띄어쓰기 없이 정답을 입력해주세요.',
              },
              minLength: { value: 1, message: '2자 이상 입력해주세요.' },
              maxLength: { value: 10, message: '10자 이하 입력해주세요.' },
            })}
            placeholder="띄어쓰기 없이 정답을 입력해주세요"
            error={!!errors.answer}
          />
```


작성해주신 코드에는 `placeholder`로 띄어쓰기 없이 정답을 입력해주세요 라고 명시하고 있습니다.
리액트훅폼 `register`의 옵션 `pattern`과 정규표현식 `/^\S*$/g`을 활용해서
사용자로부터 띄어쓰기를 받지 않도록 강제할 수도 있을것 같아요!

Reply:
[895e489](https://github.com/prgrms-web-devcourse/Team-Gaerval-Dadok-FE/pull/462/commits/895e48995ff6abdc3df8f1dcf20e92f794bc8af4) 👍🏻